### PR TITLE
Migrate to npm

### PR DESCRIPTION
**Pull request by inverno**
Fixes SonarJS#845

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,18 +5,23 @@
   "directories": {
     "lib": "lib"
   },
+  "bin": "./bin/sonarjs",
   "scripts": {
     "test": "jest",
     "clear": "rm -rf lib/*",
-    "build": "yarn clear && yarn && tsc -p ."
+    "build-mvn": "cd sonarmini && mvn clean install && cd ..",
+    "copy-jars":
```
Comment (vilchik-elena) [inline package.json]: @inverno why new lines?

File: package.json [hunk 2]
```diff
@@ -5,18 +5,23 @@
...
+      "cp sonarmini/target/sonarjs-cli-*-SNAPSHOT.jar lib/ && mkdir lib/plugins && cp sonarmini/target/plugins/* lib/plugins/",
```
Comment (vilchik-elena) [inline package.json]: @inverno imo jar should have another name

Comment (inverno) [reply package.json]: `miniscanner`

File: package.json (hunk 1 above)
Comment (inverno) [reply package.json]: @vilchik-elena the ways of prettier are mysterious

File: package.json (hunk 2 above)
Comment (vilchik-elena) [reply package.json]: @inverno `mini-scanner`?

File: .travis.yml [hunk 3]
```diff
@@ -1,26 +1,16 @@
...
+
+node_js:
+  - "8"
 
 jdk:
   - oraclejdk8
 
-install: true
+git:
+  submodules: false
```
Comment (vilchik-elena) [inline .travis.yml]: @inverno why? there is no submodules anyway

File: travis.sh [hunk 4]
```diff
@@ -1,13 +1,4 @@
 #!/bin/bash
 
-set -euo pipefail
```
Comment (vilchik-elena) [inline travis.sh]: @inverno keep this one. It should fail the build fast

File: package.json [hunk 5]
```diff
@@ -27,6 +27,7 @@
   },
   "homepage": "https://github.com/SonarSource/sonarjs-cli",
   "devDependencies": {
+    "@types/node": "^8.0.54",
```
Comment (vilchik-elena) [inline package.json:30]: @inverno what for this dependency?

File: tests/integration.test.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+import { spawn } from "child_process";
+import * as path from "path";
+
+describe("sonarjs", () => {
+  it("scans integration_test_project", done => {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
+    const sonarjs = spawn("../../bin/sonarjs", [], {
+      cwd: path.join(__dirname, "integration_test_project")
```
Comment (vilchik-elena) [inline tests/integration.test.ts:27]: @inverno don't you think that we could have a smaller file for this IT?

File: package.json (hunk 5 above)
Comment (inverno) [reply package.json:30]: @vilchik-elena ease of use of TS for the integration test

File: travis.sh (hunk 4 above)
Comment (inverno) [reply travis.sh]: In fact, I just removed `travis.sh`

**Pull request merged**